pkg/management/barman/archiver/archiver.go

```diff
@@ -148,7 +149,12 @@ func (archiver *WALArchiver) ArchiveList(
 // Archive archives a certain WAL file using barman-cloud-wal-archive.
 // See archiveWALFileList for the meaning of the parameters
 func (archiver *WALArchiver) Archive(walName string, baseOptions []string) error {
-	options := make([]string, len(baseOptions), len(baseOptions)+1)
+	optionsLength := len(baseOptions)
+	optionsLengthMax := optionsLength + 1
+	if optionsLength >= math.MaxInt || optionsLengthMax >= math.MaxInt {
```

Comment:
this condition is redundant. It is enough to check `optionsLengthMax >= math.MaxInt`. Unless the codeql thing complains about that....